prboom2/src/g_game.c

```diff
@@ -3512,21 +3578,175 @@ const byte* G_ReadDemoHeaderEx(const byte *demo_p, size_t size, unsigned int par
   demover = *demo_p++;
   longtics = 0;
 
+  // ano - jun2019 - special extensions. originally for UMAPINFO but
+  // designed to be extensible otherwise using the list of strings.
+  // note: i consulted the eternity engine implementation of this function
+  extension_version = -1;
   if (demover == 255)
   {
-	  // Uses UMAPINFO. The real version will be in the second byte.
-	  // This prepended 255 is here to prevent non-UMAPINFO ports from recognizing the demo.
-	  demover = *demo_p++;
-	  if (!umapinfo_loaded)
-	  {
-		  lprintf(LO_ERROR, "UMAPINFO not loaded but trying to play a demo recorded with it\n");
-	  }
-  }
-  else if (umapinfo_loaded)
-  {
-	  // Q: Should this abort?
-	  lprintf(LO_ERROR, "UMAPINFO loaded but trying to play a demo recorded without it\n");
+    int num_extensions;
+    // ano - jun2019
+    // so the format is
+    // demover byte == 255
+    // "PR+UM" signature (w/ ending null terminator)
+    // extension_version byte. for now this should always be "1"
+    // 2 bytes for num_extensions (little-endian)
+
+    // num_extensions *
+    //    1 byte string length
+    //    and length chars (up to 65535 obviously)
+    // note that the format has each length by each string
+    // as opposed to a table of lengths
+
+    // an example extensions string is "UMAPINFO".
+    // In no realistic scenario should num_extensions
+    // ever reach past 65535.
+
+    // so that's a total of 1+6+1+2 + (n * m) bytes + ?? for extensions
+    // or 10 + some ?? bytes + some more ??
+
+    // then finally the "real" demover byte is present here
+
+    if (CheckForOverrun(header_p, demo_p, size, 10, failonerror))
+      return NULL;
+
+    // we check for the PR+UM signature as mentioned.
+    // MBF and Eternity Engine also use 255 demover, with other signatures.
```

Comment:
I think MBF uses 203 as its demo version number. Please double-check and correct here if this is true or tell me if I'm wrong. ;)

Reply:
Fixed. :P